engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/PersistenceConnectionTest.java

```diff
@@ -0,0 +1,151 @@
+import org.camunda.bpm.engine.ProcessEnginePersistenceException;
+import org.camunda.bpm.engine.identity.UserQuery;
+import org.camunda.bpm.engine.rest.util.container.TestContainerRule;
+import org.camunda.commons.testing.ProcessEngineLoggingRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class PersistenceConnectionTest extends AbstractRestServiceTest {
+
+  @ClassRule
+  public static TestContainerRule rule = new TestContainerRule();
+
+  @Rule
+  public ProcessEngineLoggingRule loggingRule = new ProcessEngineLoggingRule()
+      .watch(REST_API);
+
+  protected static final String USER_QUERY_URL = TEST_RESOURCE_ROOT_PATH + "/user";
+
+  private final ConnectionSubclass subclass;
+
+  public PersistenceConnectionTest(ConnectionSubclass subclass) {
+    this.subclass = subclass;
+  }
+
+  @Parameters(name = "{index}: {0}")
+  public static Collection<Object[]> data() {
+    ConnectionSubclass[] values = ConnectionSubclass.values();
+
+    return Arrays.stream(values)
+        .map(c -> new Object[] {c})
+        .collect(Collectors.toList());
+  }
+
+  @Test
+  public void shouldLogError() {
+    createFailingQueryStub(subclass);
+
+    String expectedMessage = "This is an exception originating from the engine persistence layer";
+
+    given().queryParam("firstName", EXAMPLE_USER_FIRST_NAME)
+        .then().expect()
+        .statusCode(500)
+        .body("type", equalTo("ProcessEnginePersistenceException"))
+        .body("message", equalTo(expectedMessage))
+        .body("code", equalTo(0))
+        .when().get(USER_QUERY_URL);
+
+    verifyLogs(Level.ERROR, expectedMessage);
+  }
+
+  protected void verifyLogs(Level logLevel, String message) {
+    List<ILoggingEvent> logs = loggingRule.getLog();
+
+    assertThat(logs).hasSize(1);
+    assertThat(logs.get(0).getLevel()).isEqualTo(logLevel);
+    assertThat(logs.get(0).getMessage()).containsIgnoringCase(message);
+  }
+
+  static class UserQueryFactory {
+
+    static void createFailingQueryStub(ConnectionSubclass subclass) {
+      UserQuery result = mock(UserQuery.class);
+
+      when(result.list()).thenThrow(ProcessEnginePersistenceExceptionFactory.create("list", subclass));
+      when(processEngine.getIdentityService().createUserQuery()).thenReturn(result);
+    }
+  }
+
+  static class ProcessEnginePersistenceExceptionFactory {
+
+    public static ProcessEnginePersistenceException create(String operation, ConnectionSubclass sqlSubclass) {
+      SQLException rootCause = new SQLException(sqlSubclass.message(), sqlSubclass.sqlState());
+
+      return new ProcessEnginePersistenceException(
+          "This is an exception originating from the engine persistence layer",
+          new PersistenceException("Failed to execute " + operation, rootCause)
```

Comment:
❓ There is a [util class for creating ProcessEnginePersistenceExceptions](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L357). Any reason we can't use it here?

Reply:
Didn't use it because it is part of the implementation under test 🤔